packages/widgetbook/lib/src/fields/date_time_field.dart

```diff
@@ -0,0 +1,114 @@
+import 'package:flutter/material.dart';
+
+import 'field.dart';
+import 'field_codec.dart';
+import 'field_type.dart';
+
+class DateTimeField extends Field<DateTime> {
+  DateTimeField({
+    required super.name,
+    super.initialValue,
+    super.onChanged,
+    required this.readOnly,
+  }) : super(
+          type: FieldType.dateTime,
+          codec: FieldCodec<DateTime>(
+            toParam: (value) {
+              final string = asString(value);
+              // replace all instances of ':' with '~' to avoid
+              // issues with retrieving the value from the param
+              // in the query group because it is saved in a map
+              return string.replaceAll(':', timeSeparator);
+            },
+            toValue: (param) {
+              if (param == null) return null;
+              final string = param.replaceAll(timeSeparator, ':');
+              return DateTime.tryParse(string);
+            },
+          ),
+        );
+
+  /// The separator used to replace ':' in the [DateTime] string.
+  static const timeSeparator = '~';
+
+  /// The staring [DateTime] value used for the date and time pickers.
+  static final startDateTime = DateTime(0);
+
+  /// The default [DateTime] value used for the date and time pickers
+  /// and the field.
+  static DateTime get defaultDateTime => DateTime.now();
+
+  /// The ending [DateTime] value used for the date and time pickers.
+  static final endDateTime = DateTime.utc(275760, 09, 13);
```

Comment:
To simplify the implementation here, I would do the following:
1. Make `startDateTime` and `endDateTime` required class fields.*
2. Make `initialValue` required as well, so that we do not need a `defaultDateTime`.

* You will need to override `toJson` if you've added any new fields:

```dart
  @override
  Map<String, dynamic> toJson() {
    return {
      'start': codec.toParam(startDateTime),
      'end': codec.toParam(endDateTime),
    };
  }
```